src/components/CurrentUser/CurrentUser.jsx

```diff
@@ -1,12 +1,92 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { getUserId } from '../../data/api';
 import './CurrentUser.scss';
 
-export const CurrentUser = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+export class CurrentUser extends React.Component {
+  state = {
+    user: null,
+    loading: false,
+    error: false,
+  }
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+  componentDidMount() {
+    const { userId } = this.props;
+
+    this.setState({ loading: true });
+    getUserId(userId).then((response) => {
+      this.setState({
+        user: response.data,
+        loading: false,
+      });
+    }).catch(() => this.setState({
+      loading: false,
+      error: true,
+    }));
```

Comment:
You are duplicating this code twice in your code. In componentDiDMount and in ComponentDidUpdate. Let's make separate function for fetching user and use it in two places instead of repeating your code